src/commands/local.ts

```diff
@@ -49,52 +49,98 @@ export async function startLocalEnvironment(options: GenezioLocalOptions) {
-          }
 
-          // TODO: this is not very generic error handling. The SDK should throw Genezio errors, not babel.
-          if (error.code === "BABEL_PARSER_SYNTAX_ERROR") {
-            log.error("Syntax error:");
-            log.error(`Reason Code: ${error.reasonCode}`);
-            log.error(
-              `File: ${error.path}:${error.loc.line}:${error.loc.column}`
-            );
+    // eslint-disable-next-line no-async-promise-executor
+    const bundlerPromise = new Promise<any>(async (resolve, reject) => {
+      try {
+        if (yamlProjectConfiguration.classes.length === 0) {
+          throw new Error(GENEZIO_NO_CLASSES_FOUND);
+        }
+
+        sdk = await sdkGeneratorApiHandler(yamlProjectConfiguration).catch(
+          (error) => {
+            debugLogger.log("An error occured", error);
+            if (error.code === 'ENOENT') {
+              log.error(`The file ${error.path} does not exist. Please check your genezio.yaml configuration and make sure that all the file paths are correct.`)
+              throw error
+            }
+
+            // TODO: this is not very generic error handling. The SDK should throw Genezio errors, not babel.
+            if (error.code === "BABEL_PARSER_SYNTAX_ERROR") {
+              log.error("Syntax error:");
+              log.error(`Reason Code: ${error.reasonCode}`);
+              log.error(
+                `File: ${error.path}:${error.loc.line}:${error.loc.column}`
+              );
+
+              throw error;
+            }
 
             throw error;
           }
+        );
 
-          throw error;
-        }
-      );
+        projectConfiguration = new ProjectConfiguration(
+          yamlProjectConfiguration,
+          sdk
+        );
 
-      projectConfiguration = new ProjectConfiguration(
-        yamlProjectConfiguration,
-        sdk
-      );
+        processForClasses = await startProcesses(projectConfiguration, sdk);
+        resolve({
+          success:true,
+          projectConfiguration,
+          processForClasses,
+          sdk
+        });
+      } catch (error: any) {
+        log.error(error.message);
+        log.error(
+          `Fix the errors and genezio local will restart automatically. Waiting for changes...`
+        );
+        // If there was an error generating the SDK, wait for changes and try again.
+        await listenForChanges(undefined);
+        resolve({
+          success: false
+        });
+      }
+    });
 
-      processForClasses = await startProcesses(projectConfiguration, sdk);
-    } catch (error: any) {
-      log.error(error.message);
-      log.error(
-        `Fix the errors and genezio local will restart automatically. Waiting for changes...`
-      );
-      // If there was an error generating the SDK, wait for changes and try again.
-      await listenForChanges(undefined);
+    let promiseRes: any = await Promise.race([bundlerPromise, promiseListenForChanges]);
+    if (promiseRes.success === false) {
```

Comment:
Instead of naming the variable `success` maybe rename it to something that makes more sense in this context like `shouldRebundle`.